Vector/ViewController/SettingsViewController.m

```diff
@@ -2180,6 +2210,40 @@ - (void)onProfileAvatarTap:(UITapGestureRecognizer *)recognizer
     [self presentViewController:navigationController animated:YES completion:nil];
 }
 
+- (void)exportEncryptionKeys:(UITapGestureRecognizer *)recognizer
+{
+    [currentAlert dismiss:NO];
+
+    MXKEncryptionKeysExportView *exportView = [[MXKEncryptionKeysExportView alloc] initWithMatrixSession:self.mainSession];
+    currentAlert = exportView;
+
+    // Use a temporary file for the export
+    NSURL *keyFile = [NSURL fileURLWithPath:[NSTemporaryDirectory() stringByAppendingPathComponent:@"riot-keys.txt"]];
+
+    // Show the export dialog
+    __weak typeof(self) weakSelf = self;
+    [exportView showInViewController:self toExportKeysToFile:keyFile onComplete:^(BOOL success) {
+
+        if (weakSelf && success)
+        {
+             typeof(self) self = weakSelf;
+            self->currentAlert = nil;
+
+            // Let another app handling this file
+            UIActivityViewController *activityViewController = [[UIActivityViewController alloc] initWithActivityItems:@[keyFile] applicationActivities:nil];
```

Comment:
You should use UIDocumentInteractionController instead of UIActivityViewController.
UIDocumentInteractionControllerDelegate is useful to delete the file after usage.
(see RoomViewController for example of usage)